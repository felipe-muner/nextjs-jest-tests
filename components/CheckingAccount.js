export default function CheckingAccount({ checkingAmount, sentToSaving }) {
  return (
    <div>
      <h1>Checking:</h1>
      <h3>{checkingAmount}</h3>
      <button id="send-button" onClick={sentToSaving}>
        Send 100 to Saving
      </button>
    </div>
  );
}
