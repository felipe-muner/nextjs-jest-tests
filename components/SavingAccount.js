import { useState } from "react";

export default function SavingAccount({ savingAmount, sentToChecking }) {
  const [num, setNum] = useState(0);
  return (
    <div>
      <h1>Saving:</h1>
      <h3 id="saving-val">{savingAmount + "$"}</h3>
      <button id="send-to-checking" onClick={sentToChecking}>
        Send 100 to Checking
      </button>

      <button id="setnum" onClick={() => setNum(num + 1)}>
        set num
      </button>
      <div id="num-val">{num}</div>
    </div>
  );
}
