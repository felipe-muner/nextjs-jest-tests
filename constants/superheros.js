export const SUPERHEROES = ["Bat2man", "superman", "deadpool", "wolverine"];
