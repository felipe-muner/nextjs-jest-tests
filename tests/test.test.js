/*
 * @jest-environment jsdom
 */

import Index from "../pages/index";
import CheckingAccount from "../components/CheckingAccount";
import SavingAccount from "../components/SavingAccount";
import { shallow, mount } from "enzyme";
import { SUPERHEROES } from "../constants/superheros";

const user = {
  checkingAmount: 300,
};

test("H2 Amount", () => {
  const wrapper = shallow(<Index />);
  const h2s = wrapper.find("h2");
  expect(h2s).toHaveLength(1);
});

describe("rendering components", () => {
  it("render App component without crashing", () => {
    shallow(<Index />);
  });
  it("header no crashing", () => {
    const wrapper = shallow(<Index />);
    const header = <h2>My Savings</h2>;
    expect(wrapper.contains(header)).toEqual(true);
  });
  it("Button names", () => {
    const wrapper = mount(<CheckingAccount />);
    const label = wrapper.find("#send-button").text();
    expect(label).toEqual("Send 100 to Saving");
  });
});

describe("Saving account - passing props", () => {
  const savingAccount = mount(<SavingAccount savingAmount={300} />);
  it("it accepts user acc props", () => {
    expect(savingAccount.props().savingAmount).toBe(user.checkingAmount);
  });
  it("it shows same text", () => {
    const total = savingAccount.find("#saving").text();
    const expected = user.checkingAmount + "$";
    expect(expected).toEqual(total);
  });
});

describe("Simulating clicks", () => {
  it("Test click event", () => {
    const mockCallBack = jest.fn();
    const button = shallow(<button onClick={mockCallBack}>Ok!</button>);
    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe("Simulating clicks 2", () => {
  const index = mount(<Index />);
  index.find("#test-increment").simulate("click");
  const counter = index.find("#counter").text();
  expect(counter).toEqual("1");
});

describe("snapshots", () => {
  const tree = shallow(<Index />);
  it("should match snapshot", () => expect(tree).toMatchSnapshot());
});


describe("Constants", () => {
  it("SUPERHEROES should match snapshot", () => expect(SUPERHEROES).toMatchSnapshot());
});