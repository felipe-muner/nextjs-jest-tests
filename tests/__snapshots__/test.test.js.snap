// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants SUPERHEROES should match snapshot 1`] = `
Array [
  "Batman",
  "superman",
  "deadpool",
  "wolverine",
]
`;

exports[`snapshots should match snapshot 1`] = `
<Fragment>
  <div>
    <h2>
      My Savings
    </h2>
  </div>
  <button
    id="test-increment"
    onClick={[Function]}
  >
     
    increment
  </button>
  <div
    id="counter"
  >
    0
  </div>
  <div>
    <div>
      <div>
        <div>
          <CheckingAccount
            checkingAmount={3850}
            sentToSaving={[Function]}
          />
        </div>
        <div>
          <SavingAccount
            savingAmount={9820}
            sentToChecking={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
